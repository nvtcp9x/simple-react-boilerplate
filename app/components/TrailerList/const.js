export const URL_YOUTUBE = 'http://youtube.com/embed/';
