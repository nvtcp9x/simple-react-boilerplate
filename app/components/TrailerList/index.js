// @flow

import * as React from 'react';
import styled from 'styled-components'
import { URL_YOUTUBE } from './const';

const Wrapper = styled.div`
  display: flex;
  overflow: auto;
  justify-content: space-between;
`

type Props = {
  trailers: Array<Object>
}

const TrailerList = ({ trailers }: Props) => (
  <Wrapper>
    { trailers.map(trailer => <iframe title={trailer.key} src={`${URL_YOUTUBE}${trailer.key}`} allowFullScreen key={trailer.key} />) }
  </Wrapper>
)


export default TrailerList
