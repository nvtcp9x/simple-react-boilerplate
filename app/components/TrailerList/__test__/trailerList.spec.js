import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import TrailerList from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('<TrailerList />', () => {
  const mockProps = {
    trailers: [
      {
        key: 'key',
      },
    ],
  }
  const component = shallow(<TrailerList {...mockProps} />)

  it('should return div tag', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
})
