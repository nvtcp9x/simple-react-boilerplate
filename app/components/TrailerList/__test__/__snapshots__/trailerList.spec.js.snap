// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrailerList /> should return div tag 1`] = `
<styled.div>
  <iframe
    allowFullScreen={true}
    key="key"
    src="http://youtube.com/embed/key"
    title="key"
  />
</styled.div>
`;
